src/Runner.Worker/JobExtension.cs

```diff
@@ -239,9 +239,23 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
 
                             actionRunner.TryEvaluateDisplayName(contextData, context);
                             jobSteps.Add(actionRunner);
+
+                            if (prepareResult.PreStepTracker.TryGetValue(step.Id, out var preStep))
```

Comment:
nit: would it be more straight forward for PrepareResult.PreStepTracker to be a list? Then could just simply add to preJobSteps? I find myself jumping back and forth in this file trying to figure out what the dictionary lookup solves.

Reply:
I need to tracking which action.id has PreStep, so the `pre` and `main` can share same `intraActionState`